Lib/pathlib/_abc.py

```diff
@@ -895,6 +895,45 @@ def rmdir(self):
         """
         raise UnsupportedOperation(self._unsupported_msg('rmdir()'))
 
+    def rmtree(self, ignore_errors=False, on_error=None):
+        """
+        Recursively delete this directory tree.
+
+        If *ignore_errors* is true, exceptions raised from scanning the tree
+        and removing files and directories are ignored. Otherwise, if
+        *on_error* is set, it will be called to handle the error. If neither
+        *ignore_errors* nor *on_error* are set, exceptions are propagated to
+        the caller.
+        """
+        if ignore_errors:
+            def on_error(err):
+                pass
+        elif on_error is None:
+            def on_error(err):
+                raise err
+        try:
+            if self.is_symlink() or self.is_junction():
+                raise OSError("Cannot call rmtree on a symbolic link")
+            results = self.walk(
+                on_error=on_error,
+                top_down=False,
```

Comment:
Not sure if it's needed, but maybe you could say that removal is performed in a bottom-up approach.